src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -691,6 +754,21 @@ def get_eventual_alias(self):
         return None
 
 
+@ansi_dialect.segment()
+class MainTableExpressionSegment(BaseSegment):
+    """The main table expression e.g. within a FROM clause."""
+
+    type = "main_table_expression"
+    match_grammar = OneOf(
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+        Ref("TableReferenceSegment"),
+        # Nested Selects
+        Bracketed(Ref("SelectableGrammar")),
+        # Values clause?
+    )
+
```

Comment:
I think this should probably be a grammar rather than a segment. I don't think it has any structural _meaning_, it's just for easy of defining the grammar.